keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/SqlDiffFingerprintRepository.kt

```diff
@@ -82,4 +82,15 @@ class SqlDiffFingerprintRepository(
     }
     return count ?: 0
   }
+
+  override fun seen(resourceId: String, diff: ResourceDiff<*>): Boolean =
+    sqlRetry.withRetry(READ) {
+      jooq
```

Comment:
Note that jooq has a [fetchExists](https://www.jooq.org/javadoc/latest/org.jooq/org/jooq/DSLContext.html#fetchExists(org.jooq.Select)) method, which you could use to rewrite the query like this:

```
jooq.fetchExists(
  selectFrom(DIFF_FINGERPRINT)
  .where(DIFF_FINGERPRINT.RESOURCE_ID.eq(resourceId))
  .and(DIFF_FINGERPRINT.HASH.eq(diff.generateHash())))
```

Reply:
oh nice, that's  much better